### PR TITLE
Fix backing up and restoring database names containing special characters

### PR DESCRIPTION
**Pull request by chrishajas**
See individual commit messages for details.

Co-authored-by: Chris Hajas <chajas@pivotal.io>
Co-authored-by: Kevin Yeap <kyeap@pivotal.io>

### CODE REVIEW

File: integration/metadata_globals_queries_test.go [hunk 1]
```diff
@@ -40,18 +40,21 @@ var _ = Describe("backup integration tests", func() {
 	})
 	Describe("GetDatabaseInfo", func() {
 		It("returns a database info struct", func() {
-			if connectionPool.Version.Before("6") {
```
Comment (khuddlefish) [inline integration/metadata_globals_queries_test.go]: do we not need this anymore?

File: utils/io_test.go [hunk 2]
```diff
@@ -13,6 +13,38 @@ import (
...
+
+			Expect(resultString).To(Equal(`test`))
+		})
+		It("returns single quote when passed a single quote", func() {
+			dbname := `"`
+			resultString := utils.UnquoteIdent(dbname)
+
+			Expect(resultString).To(Equal(`"`))
+		})
+		It("returns empty string when passed and empty string", func() {
```
Comment (khuddlefish) [inline utils/io_test.go]: passed **an** empty string. (Don't worry about this if you aren't changing other things though)

File: utils/io_test.go [hunk 3]
```diff
@@ -13,6 +13,38 @@ import (
 )
 
 var _ = Describe("utils/io tests", func() {
+	Describe("UnquoteIdent", func() {
```
Comment (khuddlefish) [inline utils/io_test.go]: Should there be some test case for things containing single quotes? `'`

File: integration/metadata_globals_queries_test.go (hunk 1 above)
Comment (chrishajas) [reply integration/metadata_globals_queries_test.go]: We're creating that tablespace `test_tablespace`, but backing up `testdb`, which using `pg_default`. So `test_tablespace` isn't being used at all here.

File: utils/io_test.go (hunk 3 above)
Comment (chrishajas) [reply utils/io_test.go]: quote_ident doesn't care about single quotes, they don't need to be escaped or treated differently.

**Comment by chrishajas** [comment]
@khuddlefish I agree, we'll change the arg names so things are clearer.

**Pull request merged**